app/src/main/java/com/graeberj/taskman/core/presentation/TaskmanViewModel.kt

```diff
@@ -15,27 +15,33 @@ import javax.inject.Inject
 @HiltViewModel
 class TaskmanViewModel @Inject constructor(
     private val repository: AuthRepository,
-    private val preferences: Preferences,
+    private val preferences: Preferences
 ) : ViewModel() {
 
     private val _state = MutableStateFlow(TaskmanState())
     val state = _state.asStateFlow()
 
     init {
         viewModelScope.launch {
-            when (val result = repository.authenticateUser()){
+            when (repository.authenticateUser()) {
                 is AuthResult.Authorized -> {
-                    val data = result.data
                     _state.update { it.copy(isLoggedIn = true, isLoading = false) }
                 }
+
                 is AuthResult.Unauthorized -> {
                     _state.update { it.copy(isLoggedIn = false, isLoading = false) }
                 }
+
                 is AuthResult.Error -> {
-                    val errorMessage = result.message
                     _state.update { it.copy(isLoggedIn = false, isLoading = false) }
                 }
             }
         }
     }
+
+    fun onLogout() {
+        viewModelScope.launch {
+            preferences.deleteUser()
+        }
+    }
```

Comment:
Here, I'd add a `withContext(NonCancellable)` block, since otherwise this coroutine might be cancelled when you pop the backstack while deleting the user (popping the backstack would clear this VM and therefore also `viewModelScope`)